scripts/sidebarContext.js

```diff
@@ -165,5 +240,54 @@ async function updateChildren() {
             }
         }
     }).render(true)
-    
-}
\ No newline at end of file
+
+}
+
+async function resetDoors(isCurrentScene, id) {
+  if (isCurrentScene) {
+    await canvas
+      .walls?.doors.filter((item) => item.data.ds == 1)
+      .forEach((item) => item.update({ ds: 0 }, {}));
+  } else {
+    if (id) {
+      await game
+        .scenes?.get(id)
+        ?.data.walls.filter((item) => item.data.door != 0)
+        .forEach((x) => (x.data.ds = 0));
```

Comment:
Not actually updates the values in the db, just a local change from what I can tell.

Reply:
Replaced with ```
async function resetDoors(isCurrentScene, id) {
  let updates = []
  if (isCurrentScene) {
    canvas
      .walls?.doors.filter((item) => item.data.ds === 1)
      .forEach((item) => updates.push({ _id: item.id, ds: 0 }))
    await canvas.scene.updateEmbeddedDocuments("Wall", updates)
  } else {
    if (id) {
      let scene = game.scenes?.get(id)
      scene
        .data.walls.filter((item) => item.data.ds === 1)
        .forEach((x) => updates.push({ _id: x.id, ds : 0 }))
      await scene.updateEmbeddedDocuments("Wall", updates)
    }
  }
  ui.notifications?.info(`Doors have been shut.`);
}```